src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -136,29 +172,103 @@ static bool parse_timeout(grpc_json *field, grpc_millis *timeout) {
   int seconds = gpr_parse_nonnegative_int(buf);
   gpr_free(buf);
   if (seconds == -1) return false;
-  *timeout = seconds * GPR_MS_PER_SEC + nanos / GPR_NS_PER_MS;
+  *duration = seconds * GPR_MS_PER_SEC + nanos / GPR_NS_PER_MS;
+  return true;
+}
+
+static bool parse_retry_policy(grpc_json *field,
+                               retry_policy_params *retry_policy) {
+  if (field->type != GRPC_JSON_OBJECT) return false;
+  for (grpc_json *sub_field = field->child; sub_field != NULL;
+       sub_field = sub_field->next) {
+    if (sub_field->key == NULL) return false;
+    if (strcmp(sub_field->key, "maxAttempts") == 0) {
+      if (retry_policy->max_attempts != 0) return false;  // Duplicate.
+      if (sub_field->type != GRPC_JSON_NUMBER) return false;
+      retry_policy->max_attempts = gpr_parse_nonnegative_int(sub_field->value);
+      if (retry_policy->max_attempts <= 1) return false;
+      if (retry_policy->max_attempts > MAX_MAX_RETRY_ATTEMPTS) {
+        retry_policy->max_attempts = MAX_MAX_RETRY_ATTEMPTS;
```

Comment:
Add GRPC_ERROR or GRPC_INFO logging to signal that we clamped this value

Reply:
Done.